subprojects/scala/src/main/java/org/gradle/api/tasks/scala/ScalaDoc.java

```diff
@@ -51,6 +53,7 @@
     private FileCollection scalaClasspath;
     private ScalaDocOptions scalaDocOptions = new ScalaDocOptions();
     private String title;
+    private final Property<String> maxMemory = getProject().getObjects().property(String.class);
```

Comment:
We want to avoid introducing new places that use `getProject()`, so instead you'll have to do something a little different. We should introduce a public `ScalaDoc(ObjectFactory)` constructor and use `ObjectFactory.property(...)` directly.

Reply:
Makes sense, https://github.com/gradle/gradle/pull/12718/commits/aeddf3e3e20819ce6ab6bb8d21d50466bb068095 should cover this